Automattic/woocommerce-payments
# Use expired cached rate data on server error

Fixes #2124 

#### Changes proposed in this Pull Request
Use the available expired cache of currencies rates on server error, instead of returning null.

I tried at first to store the expiration time on the rates cache, but didn't like the result, the code was a bit complex to read and don't like too much the idea of having multiple structures for the same array of data.

So I ended creating a new option to store the error expiration. Opted to keep the code inline instead of creating a get/set private methods for it, cause it's only used once.

#### Testing instructions
Tests should pass. You can try it on the site, changing the code to make the expiration time shorter and/or throwing an error on the server side.

Now the cached data should be returned even if it's expired when server fails.

-------------------
